### PR TITLE
Update to use geolocation hook from checkout-resources

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Adds a loading state to the geolocation input in case the query is still loading (avoid flicker of input changing from geolocation to postal code). Also uses the hook from `checkout-resources` instead of querying the source directly.

#### How should this be manually tested?

[Workspace](https://checkoutio.myvtexprod.com/?workspace=geoload).

You should be able to see the loading state on the shipping step, which replaces the time for the input flicker.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/AddressSummary.tsx [hunk 1]
```diff
@@ -44,11 +45,15 @@ const AddressSummaryWithAddress: React.VFC = () => {
 
   const addressRules = useAddressRules()
 
+  if (selectedAddress != null && addressRules == null) {
+    return <Loading />
+  }
```
Comment (lucasecdb) [inline react/AddressSummary.tsx:50]: the reason I check for the selected address here as well is so I don't show a loading state if there isn't any address to "format" (using the address rules), which would result in a awkward experience

**Review by jeffersontuc — APPROVED** [review]

**Comment by wdsrocha** [comment]
@lucasecdb, the workspace link you shared for manual testing, is unusual. Could you double check if that's ok? It still flickers for me (but I presume that this happens because I'm in the master workspace)

**Comment by lucasecdb** [comment]
> @lucasecdb, the workspace link you shared for manual testing, is unusual. Could you double check if that's ok? It still flickers for me (but I presume that this happens because I'm in the master workspace)

hmm, that's weird. why are you on the master workspace? I added the query string `?workspace=geoload` precisely to use the workspace I configured with these changes

**Comment by wdsrocha** [comment]
> hmm, that's weird. why are you on the master workspace?

just a guess because the query string disappears after navigation (which I think is expected, but still). My pathing was:
1. click in your link
2. put an item in the cart
3. go to checkout from minicart (which skips the cart)
4. no loading state for location appears on shipping step

**Comment by lucasecdb** [comment]
@wdsrocha I see, can you try adding the query string after you go to checkout?

**Comment by wdsrocha** [comment]
@lucasecdb sure, that works, but now the component render so fast, that I can't see the loading. How to test it, given that it's SSR (afaik)?

The code LGTM tho.

**Comment by lucasecdb** [comment]
@wdsrocha we would only be able to test if there is a client-side navigation, which we will be able to do only on master after this is published I think :-/

I think something in the redirects are "broken" tho, we could take a look into it sometime, but it shouldn't affect the customer's experience, only these "edge cases" we want to test hahah

**Review by wdsrocha — APPROVED** [review]
I see, no problem then :v
